.github/workflows/build_cmake.yml

```diff
@@ -38,16 +46,14 @@ jobs:
       run: cmake -E make_directory ${{runner.workspace}}/build
 
     - run:   |
+             sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
              sudo apt update
-             sudo apt install gcc-11 g++-11 clang-12 llvm-12
+             sudo apt install ${{ matrix.compiler.pkgs }}
       shell: bash
 
     - name: Install conan
       shell: bash
-      run: |
-        python3 -m pip install --upgrade pip setuptools jinja2 cmsis-svd
-        python3 -m pip install conan
-        source ~/.profile
+      run: source ~/.profile
```

Comment:
Sorry, realised this was existing code. I didn't remember it this way.. still think it would be worthwhile using the action rather than manually installing `conan`..

Reply:
@AshleyRoll currently I use this in a small hobby project but only for playing :) 
I have updated some of my repositories and have seen that this project can also be updated and hopefully it will gain some benefits for this :)

The installation from conan will be done in CMake, the jinja2 cmsis-svd are some artifacts from myside (copy & paste error) sorry for that.

If you like we can add the github action for installing / using conan but in my opinion it is better to install this in CMake because then you can open the CMakeLists.txt everywhere and it will install conan as a requirement. I'm also a fan of doing most project related stuff in CMake. 

`~/.profile` is only a workaround (currently) to reinitialize the environement. In the future I want also get rid of this.

Reply:
The update for github actions are necessary and it was a warning in actions itself because v2 was developed with nodejs 12 which is deprecated.

catch2 was updated to latest major version because it will have some compilation speed improvements.